### PR TITLE
Ability to write test assertions

### PR DESCRIPTION
**Pull request by conor-jh**
Adds ability to attach test assertions (for status code only) when creating mockups. 

## Overview

- Adds a button below each mockup that opens a “Test Assertions” form
- Users can choose to expect a range of status codes or a specific status code
- Assertions are saved to their corresponding test object under the `expectedStatus` key as an array:
  - Exact assertions will appear like `[200]`
  - Ranges will appear with the lower bound at index 0, and the upper bound at index 1: `[200, 299]`
- `ResponseComponent`s in the dest panel will display the assertion before tests are sent
- After tests are sent, each `ResponseComponent` will display whether the test passed or failed
- If no assertion is created, we’ll assume status codes 200–299 are “good” and any others are “bad”
- If an assertion fails, the icon will change to a red x and a brief explanation will appear

### Known issues
The main issue is that there’s very little validation for range assertions. While the input has to be a number, it’s theoretically possible to switch the bounds depending on which you enter first. 

Also, validation isn’t styled right now and looks ugly if it appears.

### CODE REVIEW

File: src/main/components/ResponseComponent.jsx [hunk 1]
```diff
@@ -45,23 +45,64 @@ const Header = styled.header`
...
+  };
 
   return (
     <ResponseWrapper>
       <Header>
         <h3>{ name || `Test #${index + 1}` }</h3>
-        <Icon status={status}>{checkmark}</Icon>
+        <Icon didPass={didPass()}>{renderCheckmark()}</Icon>
         { status || 'Ready to send' }
+        <p>{ status ? renderTestResult() : renderExpectedStatus() }</p>
```
Comment (Donaldblanc) [inline src/main/components/ResponseComponent.jsx:105]: once we figure out  what we are doing with the performance metrics and how to remove the multiple calls then I think we should get accurate status codes. Currently I am only seeing 400 with 1 test.

**Review by Donaldblanc — APPROVED** [review]

File: src/main/components/ResponseComponent.jsx (hunk 1 above)
Comment (conor-jh) [reply src/main/components/ResponseComponent.jsx:105]: What are you testing exactly?

**Review by RealJoelRivera — APPROVED** [review]
Looked through the code and checked out the app. Everything looks great to me!

File: src/main/components/ResponseComponent.jsx (hunk 1 above)
Comment (Donaldblanc) [reply src/main/components/ResponseComponent.jsx:105]: I was testing the messages API from Code smith

**Pull request merged**